r/r_cromwell_setup.R

```diff
@@ -0,0 +1,148 @@
+      }
+    }
+  }
+  return(list(NULL, NULL, NULL))
+}
+
+# Get the details of the specified app
+get_app_details <- function(env, app_name) {
+  get_app_url <- glue("{env$leonardo_url}/api/google/v1/apps/{env$google_project}/{app_name}")
+
+  res <- GET(
+    url = get_app_url,
+    query = list(includeDeleted = "true", role = "creator"),
+    add_headers(Authorization = paste("Bearer", env$token))
+  )
+  if (status_code(res) == 404) {
+    return(list("DELETED", NULL, NULL))
+  } else {
+    stop_for_status(res)
+  }
+
+  result_json <- content(res, as = "parsed")
+  custom_environment_variables <- result_json$customEnvironmentVariables
+
+  list(result_json$status, custom_environment_variables$WORKSPACE_NAMESPACE, result_json$proxyUrls)
+}
+
+# Check if cromshell is installed
+validate_cromshell <- function() {
+  cat('Scanning for correct cromshell version...\n')
+  tryCatch({
+    validate_cromshell_beta()
+  }, error = function(e) {
+    validate_cromshell_alpha()
+  })
+  return
+}
+
+validate_cromshell_alpha <- function() {
+  cat('Scanning for cromshell 2 alpha..')
+  tryCatch({
+    system2('cromshell-alpha', args = 'version', stdout = TRUE, stderr = TRUE)
+    cat('cromshell-alpha found\n')
+  }, error = function(e) {
+    cat('cromshell-alpha not found\n')
+      stop(e)
+  })
+
+  return
+}
+
+validate_cromshell_beta <- function() {
+  cat('Scanning for cromshell 2 beta...\n')
+  tryCatch({
+    system2('cromshell-beta', args = 'version', stdout = TRUE, stderr = TRUE)
+    cat('cromshell-beta found')
+  }, error = function(e) {
+    stop(e)
+  })
+}
+
+# Configure Cromwell
+configure_cromwell <- function(env, proxy_url) {
+  cat("Updating cromwell config\n")
+  file_path <- file.path(path.expand("~"), ".cromshell", "cromshell_config.json")
+  configuration <- list(
+    cromwell_server = ifelse(!is.null(proxy_url), proxy_url, ""),
+    requests_timeout = 5,
+    gcloud_token_email = env$user_email,
+    referer_header_url = env$leonardo_url
+  )
+  write(toJSON(configuration, auto_unbox = TRUE, pretty = TRUE), file_path)
+}
+
+# Find the status of the app
+find_app_status <- function(env) {
+  cat('Checking status for CROMWELL app\n')
+  app_info <- check_for_app(env)
+  app_name <- app_info[[1]]
+  app_status <- app_info[[2]]
+  proxy_url <- app_info[[3]]
+
+  configure_cromwell(env, proxy_url)
+
+  if (is.null(app_name)) {
+    message('CROMWELL app does not exist. Please create cromwell server from workbench\n')
+  } else {
+    cat(sprintf('app_name=%s; app_status=%s\n', app_name, app_status))
+    message(sprintf('Existing CROMWELL app found (app_name=%s; app_status=%s).\n', app_name, app_status))
+    quit(save = "no", status = 1, runLast = FALSE)
+  }
+}
+
+main <- function() {
+  # Iteration 1: these ENV reads will throw errors if not set.
+  env <- list(
+    workspace_namespace = Sys.getenv('WORKSPACE_NAMESPACE'),
+    workspace_name = Sys.getenv('WORKSPACE_NAME'),
```

Comment:
is workspace_name used? if not, we should remove this. If we need, we should set that in our backend: https://github.com/all-of-us/workbench/blob/main/api/src/main/java/org/pmiops/workbench/leonardo/LeonardoCustomEnvVarUtils.java#L29

(I remember there was an support ticket reporting some errors about this env var). Since you are here, can you also remove this env var in python snippet if we are not using that one?

Reply:
Done